quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -296,15 +292,38 @@ public void validate(ConfigValue value) {
         }
     }
 
-    public void validateExpectedValues(ConfigValue value) {
-        if (PropertyMapperParameterConsumer.isExpectedValue(getExpectedValues(), value.getValue())) {
-            return;
+    public void validateExpectedValues(ConfigValue configValue, BiConsumer<ConfigValue, String> singleValidator) {
+        String value = configValue.getValue();
+
+        boolean multiValued = getOption().getType() == java.util.List.class;
+
+        for (String v : multiValued ? value.split(",") : new String[] { value }) {
```

Comment:
nit: I'd prefer to create separate field `var values=multiValued ? value.split(",") : new String[] { value }` and then only using `for (String v : values) {`. But it's only a matter of preferences, keep it as is if you want :))

Reply:
Added the intermediate variable.